CONTRIBUTING.md

```diff
@@ -0,0 +1,273 @@
+
+- Output of `salt --versions-report`
+- Output of `bootstrap-salt.sh -v`
+- System type
+- Cloud/VM provider as appropriate
+
+Valid bugs will worked on as quickly as resources can be reasonably allocated.
+
+### Features
+
+Feature additions and requests are welcomed. When requesting a feature it will
+be placed under the `Feature` label.
+
+If a new feature is desired, the fastest way to get it into Salt Bootstrap is
+to contribute the code. Before starting on a new feature, an issue should be
+filed for it. The one requesting the feature will be able to then discuss the
+feature with the Salt Bootstrap maintainers and discover the best way to get
+the feature included into the bootstrap script and if the feature makes sense.
+
+It is possible that the desired feature has already been completed.
+Look for it in the [README](https://github.com/saltstack/salt-bootstrap/blob/develop/README.rst)
+or exploring the wide list of options detailed at the top of the script. These
+options are also available by running the `-h` help option for the script. It
+is also common that the problem which would be solved by the new feature can be
+easily solved another way, which is a great reason to ask first.
+
+## Fixing Issues
+
+Fixes for issues are very welcome!
+
+Once you've fixed the issue you have in hand, create a 
+[pull request](https://help.github.com/articles/creating-a-pull-request/).
+
+Salt Bootstrap maintainers will review your fix. If everything is OK and all
+tests pass, you fix will be merged into Salt Bootstrap's code.
+
+### Branches
+
+There are two main branches in the Salt Bootstrap repository:
+
+- develop
+- stable
+
+All fixes and features should be submitted to the `develop` branch. The `stable`
+branch only contains released versions of the bootstrap script.
+
+## Pull Requests
+
+The Salt Bootstrap repo has several pull request checks that must pass before
+a bug fix or feature implementation can be merged in.
+
+### PR Tests
+
+There are several Jenkins jobs that run on each Pull Request. Most of these are
+CI jobs that set up different steps, such as setting up the job, cloning the
+repo from the PR, etc.
+
+#### Lint Check
+
+The pull request test that matters the most, and the contributor is directly 
+responsible for fixing, is the Lint check. This check *must* be passing before
+the contribution can be merged into the codebase.
+
+If the lint check has failed on your pull request, you can view the errors by
+clicking `Details` in the test run output. Then, click the `Violations` link on
+the left side. There you will see a list of files that have errors. By clicking
+on the file, you will see `!` icons on the affected line. Hovering over the `!`
+icons will explain what the issue is.
+
+To run the lint tests locally before submitting a pull request, use the
+`tests/runtests.py` file. The `-L` option runs the lint check:
+
+```
+python tests/runtests.py -L
```

Comment:
I know that test suite haven't been touched for ages. Maybe it would be better to include little instruction on how to install and run `shellcheck` instead?

Reply:
This is what the actual lint check runs in the Jenkins tests, so I would rather keep this consistent with what we're running for tests. 

I know the test suite hasn't been touched in a long time. I'm hoping to come back to that in the near future and get some more comprehensive tests running.

Reply:
Okay, sounds right. This PR is good to go from my PoV.